apps/meteor/client/views/room/body/hooks/useGetMore.ts

```diff
@@ -1,17 +1,28 @@
 import type { MutableRefObject } from 'react';
-import { useCallback } from 'react';
+import { useCallback, useRef } from 'react';
 
 import { RoomHistoryManager } from '../../../../../app/ui-utils/client';
 import { withThrottling } from '../../../../../lib/utils/highOrderFunctions';
+import { useToggleSelectAll } from '../../MessageList/contexts/SelectedMessagesContext';
 
 export const useGetMore = (rid: string, atBottomRef: MutableRefObject<boolean>) => {
+	const ref: MutableRefObject<HTMLElement | null> = useRef(null);
+	const handleToggleAll = useToggleSelectAll();
+
+	const handleGetMore = () => {
+		ref.current?.scrollTo({ top: 0, behavior: 'smooth' });
+		handleToggleAll();
+	};
```

Comment:
Instead of defining this function here in order to get the ref, I recommend making a hook for this (or, place the scroll logic in the `useToggleSelectAll`  hook) and pass the ref to the hook from the `RoomBody` component.

Reply:
also, `handleGetMore` is a little confusing, to be honest... Can we have a more descriptive name? Like `selectAllAndScrollToTop` or maybe something else.